File: 백준/Silver/1475. 방 번호/방 번호.py
import sys

N = list(sys.stdin.readline().strip())

count = [0] * 10

for i in N:
    if i == '6' or i == '9':
        if count[6] < count[9]:
            count[6] += 1
        else:
            count[9] += 1
    else:
        count[int(i)] += 1

print(max(count))

File: 백준/Silver/1475. 방 번호/방 번호.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        String s = bf.readLine();
        char[] c = s.toCharArray();

        int[] num = new int[10];
        for(int i=0; i<s.length(); i++){
            if(c[i] - 48 == 6 || c[i] - 48 == 9){
                num[9]++;
            }else{
                num[c[i] - 48]++;
            }
        }

        int max = 0;
        for(int i=0; i<9; i++){
            max = Math.max(max, num[i]);
        }

        int nine = 0;
        if(num[9] % 2 == 0) nine = num[9] / 2;
        else nine = num[9] / 2 + 1;

        System.out.println(Math.max(max, nine));


    }
}